tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/SQS/ContextPropagation.cs

```diff
@@ -26,8 +26,21 @@ private static void Inject<TMessageRequest>(SpanContext context, IDictionary mes
             // Consolidate headers into one JSON object with <header_name>:<value>
             var sb = Util.StringBuilderCache.Acquire(Util.StringBuilderCache.MaxBuilderSize);
             sb.Append('{');
+
             SpanContextPropagator.Instance.Inject(context, sb, default(StringBuilderCarrierSetter));
-            dataStreamsManager?.InjectPathwayContext(context.PathwayContext, AwsSqsHeadersAdapters.GetInjectionAdapter(sb));
+
+            if (dataStreamsManager != null && dataStreamsManager.IsEnabled && context.PathwayContext != null)
+            {
+                var base64EncodedContext = Convert.ToBase64String(PathwayContextEncoder.Encode(context.PathwayContext.Value));
+                sb.AppendFormat("\"{0}\":\"{1}\",", DataStreamsPropagationHeaders.PropagationKeyBase64, base64EncodedContext);
+
+                if (Tracer.Instance.Settings.IsDataStreamsLegacyHeadersEnabled)
+                {
+                    var binaryEncodedContext = Convert.ToBase64String(PathwayContextEncoder.Encode(context.PathwayContext.Value));
```

Comment:
Isn't this injecting the same as above by using `Convert.ToBase64String`?

Reply:
If that's intentional, then you can save the result of `Convert.ToBase64String(PathwayContextEncoder.Encode(context.PathwayContext.Value))` and re-use it to avoid the extra work and heap allocations.

Reply:
Fixed to inject different values (binary and base64) based on header